components/EmailValidator.vue

```diff
@@ -120,15 +122,30 @@ export default {
           }
         }
 
-        const validate = await this.$refs.email?.validate()
-        if (validate) {
-          this.$emit('update:valid', validate.valid)
+        const meta = this.$refs.form?.getMeta()
+        if (meta) {
+          this.$emit('update:valid', meta.valid)
         }
       },
     },
   },
   methods: {
-    validateEmail(value) {
+    async checkValidDomain(value) {
+      let isValidDomain = true
+      try {
+        const domain = value.substring(value.indexOf('@') + 1)
+        const url = new URL('https://dns.google/resolve')
+        url.search = new URLSearchParams({ name: domain }).toString()
+        const googleResponse = await fetch(url).then((response) => response)
```

Comment:
I'm slightly worried here about whether there are ad blockers which would cause this fetch never to complete.  If that happened, we'd never validate the address.  

Please can we add a timeout, say 10 seconds?  Then if the fetch doesn't complete we should just assume that the email is valid.

Reply:
I've added the new way of cancelling request using this API https://caniuse.com/mdn-api_abortsignal_timeout_static, but I'm not sure if that's okay in terms of browser support. If not, let me know and I'll rewrite with default setTimeout and AbortController to cancel the request)